addon/components/pie-chart.js

```diff
@@ -410,11 +446,18 @@ const PieChartComponent = ChartComponent.extend(FloatingTooltipMixin,
           var labelYPos = f(y);
           var labelHeight = this.getBBox().height;
           var side = labelXPos > 0 ? 'right' : 'left';
+
+          // Adjusts labelXPos after labelYPos is moved if labels overlap
+          var updateXPos = function(labelYPos) {
+            return Math.sqrt(Math.max(Math.pow(labelRadius,2) - Math.pow(labelYPos,2), 0));
```

Comment:
Wondering if this function could take in no arguments and actually set `labelXPos` on its own, because it only gets called in this scope. ie,

```
var updateXPos = function() {
   labelyXPos = Math.sqrt(Math.max(Math.pow(labelRadius,2) - Math.pow(labelYPos,2), ));
}
```

Also, could you explain a bit more what this function does, specifically why we do a `Math.max` between the pythagorean result and 0?